sparse/coo/umath.py

```diff
@@ -150,7 +146,7 @@ def _elemwise_n_ary(func, *args, **kwargs):
 
     args = list(args)
 
-    args_zeros = tuple(_zero_of_dtype(arg.dtype)[()] for arg in args)
+    args_zeros = tuple(_zero_of_dtype(np.dtype(arg)) for arg in args)
```

Comment:
Remove the indexing to get a 0D array instead of a scalar, some functions work only on arrays.